Provide warning message for unnecessary sudo

djmgit opened the pull request:
Fixes #4420. If minikube is started with root privilege and vm-driver is
not hyperV or none, then a warning message is displayed.

djmgit commented:
CLI has been signed.

djmgit commented:
/assign @tstromberg

djmgit commented:
I signed it

medyagh commented on cmd/minikube/cmd/start.go:
```diff
@@ -172,6 +173,13 @@ func runStart(cmd *cobra.Command, args []string) {
+	currentUser, err := user.Current()
+
+	// Display warning if minikube is being started with root and vmDriver is not hyperv
+	if err == nil && currentUser.Name == "root" && viper.GetString(vmDriver) != "hyperv" {
```
could you check if this works under windows too ? is it called "root" or "admin" or "adminstrator" in windows?

afbjorklund replied:
Using sudo (or running as root) is required, when using the `none` driver (as a special case).

medyagh commented on cmd/minikube/cmd/start.go:
```diff
@@ -172,6 +173,13 @@ func runStart(cmd *cobra.Command, args []string) {
 	console.OutStyle(console.Happy, "minikube %s on %s (%s)", version.GetVersion(), runtime.GOOS, runtime.GOARCH)
 	validateConfig()
 
+	currentUser, err := user.Current()
```
could you log out the err if it no nil ? 
like "error getting the current user" (without failing) and there is an error getting the current user, skip the warning ... ( we dont want to fail minikube becaause we can't warn them about which user is best to run minikube

medyagh commented:
Thanks for making this PR, I am excited to review the PR. could you please sign the CLA?

djmgit commented:
@medyagh I will make the changes you suggested and update my PR soon.
I have already signed the CLA, however its not reflecting here. Should I close this PR and create a new one?

medyagh commented:
> @medyagh I will make the changes you suggested and update my PR soon.
> I have already signed the CLA, however its not reflecting here. Should I close this PR and create a new one?

no you don't need to create the a new PR, just make sure to use the same email for git that is used in linux foundation. (https://help.github.com/en/articles/setting-your-commit-email-address-in-git)
 you need to re-base your PR to make a commit message vaild. the current commit message is  not valid because it has "Issue : #4420 " in it.

djmgit commented:
@afbjorklund cool, will add that case too

djmgit commented:
I signed it

djmgit commented:
@medyagh I have updated my PR to log a message when err is not nil. Also I have added the case mentioned by @afbjorklund 
I tested golang's user.Current() on windows. The problem with windows is, when the user.Current() runs with admin privilege (run as administrator), currentUser.Name does not return root. Rather it returns the name of the user who is running the code with admin privilege. So in windows, just the return value of currentUser.Name I guess does not ensure whether it is running with admin privileges or not

medyagh commented:
@minikube-bot okay to test

sharifelgamal commented:
Yeah, testing whether a Windows users has raised privileges is a little bit tricky, but there are ways to do it.

djmgit commented:
@medyagh I have already considered none driver case after @afbjorklund pointed it out.

djmgit commented:
@medyagh is there anything more needed in this PR?

medyagh commented on cmd/minikube/cmd/start.go:
```diff
@@ -172,6 +173,15 @@ func runStart(cmd *cobra.Command, args []string) {
+
+	// Display warning if minikube is being started with root and vmDriver is not HyperV
+	if err != nil {
+		fmt.Println("Error getting the current user")
```
Please use glog.Errorf

djmgit commented:
@medyagh done

medyagh commented on cmd/minikube/cmd/start.go:
```diff
@@ -172,6 +173,15 @@ func runStart(cmd *cobra.Command, args []string) {
+
+	// Display warning if minikube is being started with root and vmDriver is not HyperV
+	if err != nil {
+		glog.Errorln("Error getting the current user: ", err)
```
please always use %v for errror.
like glog.Errorf(""Error getting the current user %v", err)

medyagh commented:
@minikube-bot OK to test

djmgit commented:
@medyagh should I update it, or is it alright?
Actually I saw some examples in the codebase where glog.Errorln was used so I used like that.

medyagh commented:
> @medyagh should I update it, or is it alright?
> Actually I saw some examples in the codebase where glog.Errorln was used so I used like that.

for err it is always better to use %v, lets use %v for err

djmgit commented:
@medyagh done

medyagh commented:
Awesome thanks for the contribution and your patience to make quality code on minikube, It looks good to me after the tests run.

medyagh commented:
@minikube-bot OK to test
